src/gnu/trove/array/AbstractOffheapArray.java

```diff
@@ -53,11 +60,41 @@ public AbstractOffheapArray(long capacity) {
         }
         address = UNSAFE.allocateMemory(capacity);
         boxedAddress = new AtomicLong(address);
-        cleaner = Cleaner.create(this, new Deallocator(boxedAddress));
+        cleaner = createCleaner(this, new Deallocator(boxedAddress));
         UNSAFE.setMemory(address, capacity, (byte) 0);
         this.capacity = capacity;
     }
 
+    private static Object createCleaner(Object obj, Runnable cleanUpRunnable) {
+        try {
+            Class<?> cleaner = Class.forName("sun.misc.Cleaner");
+            Method create = cleaner.getMethod("create", Object.class, Runnable.class);
+            return create.invoke(null, obj, cleanUpRunnable);
+        } catch (ClassNotFoundException e) {
+            // means we are on java 9+
+        } catch (NoSuchMethodException e) {
+            // shouldn't happen, will try using java 9+ cleaner anyway
+        } catch (IllegalAccessException e) {
+            // shouldn't happen, will try using java 9+ cleaner anyway
+        } catch (InvocationTargetException e) {
+            // shouldn't happen, will try using java 9+ cleaner anyway
+        }
```

Comment:
ReflectiveOperationException?

Reply:
It is currently set to Java 6 compatibility, but I guess I don't need to care about that anymore, so I can probably do this safely.